### DISCUSSION TITLE
ISR in appDir is too slow compared to version 12

### DISCUSSION THREAD

**Discussion by guanweisong** [vercel/next.js#47636] [Help]
### Summary

I have an application based on the appDir of the v13 version and the v12 version. They share a set of api services, and both have enabled the ISR feature, and both deployed in Vercel, but the rendering speed is very different. Here are some screenshots and other infos:

nextjs version: 13.2.4
online preview address: https://v2.guanweisong.com/list/category
source address: https://github.com/guanweisong/honeycomb-wap
![image](https://user-images.githubusercontent.com/13727614/228398482-4a52d914-7864-481d-8bce-d9cbc8f4d8c9.png)

nextjs version: 12.3.1
online preview address: https://www.guanweisong.com/list/category
source address: https://github.com/guanweisong/honeycomb-wap/tree/feature/nextjs12
![image](https://user-images.githubusercontent.com/13727614/228398893-03d1245e-10b5-4967-a0f8-8ed59b3562c0.png)

This difference has puzzled me for a long time

**Comment by tonyabracadabra** [comment]
what components are you using?

**Comment by guanweisong** [comment]
@tonyabracadabra
 I guess there are two problems with the ISR in AppDir.
1. When I use generateStaticParams, the server component will render twice. See related discussion [here](https://github.com/vercel/next.js/issues/48722) and [here](https://github.com/vercel/next.js/issues/43254). This problem caused all server requests to take twice as long.
2. During prefetch, when different slugs on the same page are revalidated at the same time, the asynchronous data obtained by the same component is not shared, resulting in multiple requests for interfaces such as obtaining menu data at the same time. See related discussion [here](https://github.com/vercel/next.js/discussions/47070).
